### PR TITLE
[NanoAOD] PileupJetId speed up

### PR DESCRIPTION
**Pull request by nurfikri89**
#### PR description:

This PR is to address the issue (cms-nanoAOD#586) regarding the slow speed of pileupJetId calculation in NanoAOD production. Now, only one instance of `PileupJetIdProducer` is scheduled and configured according to era. No changes in output is expected.

Below is a comparison of the cpu usage between `CMSSW_12_3_0_pre4` and this PR using a TTJets ULMiniAODv2 sample as input with 5000 events. 

**Default (`CMSSW_12_3_0_pre4`)**
circle_nanodefault

**This PR**
circle_nanomod_puidcleanup

#### PR validation:

passes the usual runTheMatrix test: `runTheMatrix.py -l limited -i all --ibeos`

#### Backports:
This PR should be backported 10_6_X  if there is a plan for Ultra-Legacy NanoAODv10 production in the future.

### CODE REVIEW

**Comment by mariadalfonso** [comment]
enable profiling

File: PhysicsTools/NanoAOD/python/jets_cff.py [hunk 1]
```diff
@@ -269,20 +261,21 @@
...
-run2_jme_2017.toModify( jetTable.variables, puId = Var("userInt('puId106XUL17Id')", int,doc="Pileup ID flags with 106X (2017) training"))
-run2_jme_2017.toModify( jetTable.variables, puIdDisc = Var("userFloat('puId106XUL17Disc')", float,doc="Pileup ID discriminant with 106X (2017) training",precision=10))
+
+run2_jme_2017.toModify( jetTable.variables.puIdDisc, doc="Pileup ID discriminant with 106X (2017) training")
+run2_jme_2017.toModify( jetTable.variables.puId, doc="Pileup ID flags with 106X (2017) training")
+(run2_jme_2016 & ~tracker_apv_vfp30_2016 ).toModify( jetTable.variables.puIdDisc, doc="Pileup ID discriminant with 106X (2016) training")
+(run2_jme_2016 & ~tracker_apv_vfp30_2016 ).toModify( jetTable.variables.puId, doc="Pileup ID flags with 106X (2016) training")
+(run2_jme_2016 & tracker_apv_vfp30_2016 ).toModify( jetTable.variables.puIdDisc, doc="Pileup ID discriminant with 106X (2016APV) training")
+(run2_jme_2016 & tracker_apv_vfp30_2016 ).toModify( jetTable.variables.puId,  doc="Pileup ID flags with 106X (2016APV) training")
+run2_nanoAOD_102Xv1.toModify( jetTable.variables.puIdDisc, doc="Pileup ID discriminant with 102X (2018) training")
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/python/jets_cff.py:271]: I suspect that changing order here is not good and lead to changes in the 102Xv1 and 2017 era.

**Comment by mariadalfonso** [comment]
please test

File: PhysicsTools/NanoAOD/python/jets_cff.py (hunk 1 above)
Comment (nurfikri89) [reply PhysicsTools/NanoAOD/python/jets_cff.py:271]: Will revert to the original ordering in the next commit.

**Comment by mariadalfonso** [comment]
please test

**Comment by mariadalfonso** [comment]
+xpog

re-arranged sequences with modifiers
no change in the output observed

**Comment by qliphy** [comment]
+1

**Comment by nurfikri89** [comment]
@mariadalfonso Should this be backported?

**Comment by mariadalfonso** [comment]
> @mariadalfonso Should this be backported?

@nurfikri89 it's not needed for now, usually we do not backport speed up and next V10 Run2 nano production will be likely made with master

**Pull request merged**